internal/cli/cluster/connectinfo.go

```diff
@@ -0,0 +1,316 @@
+				} else if goOS == "windows" {
+					goOS = "Windows"
+				}
+				if goOS != "" && goOS != "linux" {
+					for id, value := range operatingSystemList {
+						if strings.Contains(value, goOS) {
+							operatingSystemValueWithFlag := operatingSystemList[id] + " (Detected)"
+							operatingSystemList = append([]string{operatingSystemValueWithFlag}, append(operatingSystemList[:id], operatingSystemList[id+1:]...)...)
+							break
+						}
+					}
+				}
+
+				// Get operating system
+				operatingSystemCombination, err := cloud.GetSelectedConnectOs(operatingSystemList)
+				if err != nil {
+					return err
+				}
+				operatingSystem = strings.Split(operatingSystemCombination, "/")[0]
+
+			} else { // non-interactive mode
+				clusterID, err = cmd.Flags().GetString(flag.ClusterID)
+				if err != nil {
+					return err
+				}
+
+				projectID, err = cmd.Flags().GetString(flag.ProjectID)
+				if err != nil {
+					return err
+				}
+
+				clientName, err = cmd.Flags().GetString(flag.ClientName)
+				if err != nil {
+					return err
+				}
+				if !contains(clientName, connectClientsList) {
+					return errors.New(fmt.Sprintf("Unsupported client. Run \"%[1]s cluster connect-info -h\" to check supported clients list", config.CliName))
+				}
+
+				operatingSystem, err = cmd.Flags().GetString(flag.OperatingSystem)
+				if err != nil {
+					return err
+				}
+				if !contains(operatingSystem, operatingSystemListForHelp) {
+					return errors.New(fmt.Sprintf("Unsupported operating system. Run \"%[1]s cluster connect-info -h\" to check supported operating systems list", config.CliName))
+				}
+			}
+
+			// Get cluster info
+			params := clusterApi.NewGetClusterParams().WithProjectID(projectID).WithClusterID(clusterID)
+			clusterInfo, err := d.GetCluster(params)
+			if err != nil {
+				return err
+			}
+
+			// Resolve cluster information
+			// Get connect parameter
+			defaultUser := clusterInfo.Payload.Status.ConnectionStrings.DefaultUser
+			host := clusterInfo.Payload.Status.ConnectionStrings.Standard.Host
+			port := strconv.Itoa(int(clusterInfo.Payload.Status.ConnectionStrings.Standard.Port))
+			clusterType := clusterInfo.Payload.ClusterType
+			if clusterType == DEVELOPER {
+				clusterType = SERVERLESS
+			}
+
+			// Get connection string
+			connectInfo, err := cloud.RetrieveConnectInfo(d)
+			if err != nil {
+				return err
+			}
+			connectionString, err := generateConnectionString(connectInfo, clientName, host, defaultUser, port, clusterType, operatingSystem)
+			if err != nil {
+				return err
+			}
+			fmt.Fprintln(h.IOStreams.Out, color.BlueString("Connection string"))
+			fmt.Fprintln(h.IOStreams.Out, connectionString)
+
+			return nil
+		},
+	}
+
+	cmd.Flags().StringP(flag.ProjectID, flag.ProjectIDShort, "", "Project ID")
+	cmd.Flags().StringP(flag.ClusterID, flag.ClusterIDShort, "", "Cluster ID")
+	cmd.Flags().String(flag.ClientName, "", "Connected client. Supported clients: ["+strings.Join(connectClientsList, ", ")+"]")
```

Comment:
You could also use %v; it's the same as ["+strings.Join(connectClientsList, ", ")+"]", but not that wordy。

Reply:
@xuanyu66 Thank you very much. I will try `%v`.